### PR TITLE
Run generated Ruby unit tests in CI

### PR DESCRIPTION
**Pull request by andreamlin**

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL

File: .circleci/config.yml [hunk 1]
```diff
@@ -137,6 +123,39 @@ jobs:
...
+      - attach_workspace:
+          # Use the workspace created in generate-clients
+          at: workspace
+      - run:
+          name: Install Ruby client dependencies.
+          command: |
+            gem install bundler
+            git clone https://github.com/GoogleCloudPlatform/google-cloud-ruby
+            cd google-cloud-ruby
+            bundle update
```
Comment (landrito) [inline .circleci/config.yml]: Can you explain this step?

File: .circleci/config.yml [hunk 2]
```diff
@@ -21,6 +11,12 @@ anchor_test_python: &anchor_test_python
     # https://github.com/googleapis/gapic-generator/issues/2208
   when: always
 
+anchor_test_ruby: &anchor_test_ruby
+  command: |
+    cd workspace/gapic-generator/artman-genfiles/ruby/google-cloud-ruby/${TEST_API}
+    bundle update
+    bundle exec rake test
```
Comment (landrito) [inline .circleci/config.yml]: Did we also want smoke tests?

Comment (landrito) [reply .circleci/config.yml]: `bundle exec rake acceptance`

https://github.com/googleapis/gapic-generator/blob/master/src/test/java/com/google/api/codegen/testdata/ruby/ruby_library.baseline#L414

File: .circleci/config.yml (hunk 1 above)
Comment (andreamlin) [reply .circleci/config.yml]: Ok, removed. 
(I had copypasting the steps in the Cloud Release Playbook)

**Comment by andreamlin** [comment]
PTAL

File: .circleci/config.yml [hunk 3]
```diff
@@ -137,6 +123,36 @@ jobs:
...
+    docker:
+      - image: circleci/ruby:2.4-node
+    steps:
+      - attach_workspace:
+          # Use the workspace created in generate-clients
+          at: workspace
+      - run:
+          name: Install Ruby client dependencies.
+          command: |
+            gem install bundler
```
Comment (landrito) [inline .circleci/config.yml]: Bundler is usually pre-installed on ruby docker images.

Comment (andreamlin) [reply .circleci/config.yml]: great! removed

**Pull request merged**